### ISSUE TITLE
hardhat test ignores reason string when running on localhost

### ISSUE THREAD

**Issue by BKopyIo** [NomicFoundation/hardhat#3055]
I get the following message running hh test --network localhost:

```
Error: missing revert data in call exception; Transaction reverted without a reason string [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (data="0x", transaction={"from":"0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266","to":"0x3FE03e949D365126DB640Ae1696916261F9052e0","data":"0x4e79eeac0000000000000000000000000000000000000000000000000000000000000002","accessList":null}, error={"name":"ProviderError","code":-32603,"_isProviderError":true}, code=CALL_EXCEPTION, version=providers/5.6.8)
-->
```

But on the termnal when hardhat node is running, the revert string is displayed.  
```
eth_call
  Contract call:       <UnrecognizedContract>
  From:                0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266
  To:                  0x3fe03e949d365126db640ae1696916261f9052e0

  Error: VM Exception while processing transaction: reverted with reason string '{XJU7D7}'
```

It's already pretty tough to debug errors within the solidity conracts -- would be good that whatevererror info is available is passed back to to the terminal where tests are running.

**Comment by feuGeneA** [comment]
Thank you for using Hardhat. We do have existing tests that exercise reverts with reason strings while using Hardhat Node. I'm wondering what's different about your situation.

In general, if there is a problem in Hardhat, we will need a minimal reproducible example in order to investigate it. See https://stackoverflow.com/help/minimal-reproducible-example

Can you provide us with a minimal reproducible example?

**Comment by BKopyIo** [comment]
Unfortunately I am a bit too busy sorting out the bugs in my code to spend  a considerable amount of time constructing a MRE to demonstrate that the bugs in your code are real.  I've reproduced the actual output I received, and while I get it may not be easy to troubleshoot without somebody having done the work of producing an MRE, that doesn't mean the problem doesn't exist and isn't worth at least 'minimally investigating' on your end about what might be causing such a thing, which I take it from your note isnt deemed worth the trouble.

**Comment by feuGeneA** [comment]
Our own resources are limited as well, so we need to ask users for help when possible/practical. Thanks for understanding, and thanks again for using Hardhat. I'm re-opening this in the hope that it might bubble up through our prioritization process.

**Comment by gnarvaja** [comment]
Not sure if it's the same problem, but I had a similar problem. 

Some of my tests that previously were working fine with tests like `.to.be.revertedWith("foo bar")` started to fail after moving the code to Hardhat 2.10.

```AssertionError: Expected transaction to be reverted with reason 'foo bar', but it reverted without a reason```

It was weird because some of the revert tests with reason were working fine, but others don't.

I fixed the problem fixing ethers at version 5.6.1 thanks to https://github.com/ethers-io/ethers.js/discussions/2849#discussioncomment-2701990

**Comment by fvictorio** [comment]
Closing this in favor of #1222, which should improve this kind of thing. Also: the error message is there, but the error constructed by ethers makes it hard to see.
